Build Scripts: `npm run dev` fails when the Gutenberg directory path contains spaces

### Description

Running `npm run dev` fails when the Gutenberg repository is located inside a directory whose path contains spaces.

For example, cloning Gutenberg into a LocalWP site under `Local Sites` causes the development build to fail during the package cleaning step.

The failure occurs because the build scripts invoke child processes using `shell: true`, and the script paths are not quoted before being passed to the shell. As a result, the shell splits the path at the first space and attempts to execute an incomplete path.


### Step-by-step reproduction instructions


1. Clone the Gutenberg repository into a directory containing a space.

Example:

```text
/Users/<username>/Local Sites/gutenberg/app/public/wp-content/plugins/gutenberg
```

2. Install dependencies.

```bash
npm install
composer install
```

3. Start the development build.

```bash
npm run dev
```

### Expected behavior

The development build should start successfully regardless of whether the project path contains spaces.

### Actual behavior

The build fails during the cleaning step with an error similar to:

```text
🧹 Cleaning packages...

Error: Cannot find module '/Users/<username>/Local'

Node.js v22.x.x

❌ Dev build failed:
Command failed:
node /Users/<username>/Local Sites/.../clean.mjs --packages
```

The shell interprets:

```text
/Users/<username>/Local
```

as the executable instead of the full script path.


### Screenshots, screen recording, code snippet

> gutenberg@23.6.0-rc.1 dev
> npm run --workspace @wordpress/build-scripts dev --


> @wordpress/build-scripts@0.0.0 dev
> node ./dev.mjs

🔨 Starting development build...

🔍 Checking dependencies...

   ✔ All good.

🧹 Cleaning packages...
node:internal/modules/cjs/loader:1404
  throw err;
  ^

Error: Cannot find module '/Users/<username>/Local'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1401:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1057:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1062:22)
    at Function._load (node:internal/modules/cjs/loader:1211:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:171:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.17.1

❌ Dev build failed: Command failed: node /Users/<username>/Local Sites/gutenberg/app/public/wp-content/plugins/gutenberg/tools/build-scripts/clean.mjs --packages
npm error Lifecycle script `dev` failed with error:
npm error code 1
npm error path /Users/<username>/Local Sites/gutenberg/app/public/wp-content/plugins/gutenberg/tools/build-scripts
npm error workspace @wordpress/build-scripts@0.0.0
npm error location /Users/<username>/Local Sites/gutenberg/app/public/wp-content/plugins/gutenberg/tools/build-scripts
npm error command failed
npm error command sh -c node ./dev.mjs

### Environment info

* Gutenberg: 23.6.0-rc.1
* macOS
* LocalWP (repository located inside `Local Sites`)


### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

--- a/tools/build-scripts/dev.mjs
+++ b/tools/build-scripts/dev.mjs
@@ -160,7 +160,7 @@ async function dev() {
 		console.log( '\n🧹 Cleaning packages...' );
 		await exec(
 			'node',
-			[ path.join( __dirname, 'clean.mjs' ), '--packages' ],
+			[ `"${ path.join( __dirname, 'clean.mjs' ) }"`, '--packages' ],
 			{ silent: true }
 		);
 
@@ -174,7 +174,7 @@ async function dev() {
 		// This must happen before TypeScript compilation because some packages
 		// (like vips) have source files that import from generated worker-code.ts
 		await exec( 'node', [
-			path.join( __dirname, 'packages/generate-worker-placeholders.mjs' ),
+			`"${ path.join( __dirname, 'packages/generate-worker-placeholders.mjs' ) }"`,
 		] );
 
 		if ( ! skipTypes ) {
@@ -191,16 +191,16 @@ async function dev() {
 
 			console.log( '\n✅ Checking type declaration files...' );
 			await exec( 'node', [
-				path.join(
+				`"${ path.join(
 					__dirname,
 					'packages/check-build-type-declaration-files.cjs'
-				),
+				) }"`,
 			] );
 		}
 
 		console.log( '\n📦 Building vendor files...' );
 		await exec( 'node', [
-			path.join( __dirname, 'packages/build-vendors.mjs' ),
+			`"${ path.join( __dirname, 'packages/build-vendors.mjs' ) }"`,
 		] );
 
 		const setupTime = Date.now() - startTime;

--- a/tools/build-scripts/dev.mjs
+++ b/tools/build-scripts/dev.mjs
@@ -3,7 +3,8 @@
 /**
  * External dependencies
  */
-import { execSync, spawn } from 'child_process';
+import { execSync } from 'child_process';
+import spawn from 'cross-spawn';
 import { fileURLToPath } from 'url';
 import { parseArgs } from 'util';
 import path from 'path';
@@ -29,7 +30,6 @@ function exec( command, args = [], options = {} ) {
 		const childOptions = {
 			cwd: ROOT_DIR,
 			stdio: silent ? 'pipe' : 'inherit',
-			shell: true,
 			...spawnOptions,
 		};
 
@@ -90,7 +90,6 @@ function execAsync( command, args = [], options = {} ) {
 	return spawn( command, args, {
 		cwd: ROOT_DIR,
 		stdio: 'inherit',
-		shell: true,
 		...options,
 	} );
 }
@@ -160,7 +159,7 @@ async function dev() {
 		console.log( '\n🧹 Cleaning packages...' );
 		await exec(
 			'node',
-			[ `"${ path.join( __dirname, 'clean.mjs' ) }"`, '--packages' ],
+			[ path.join( __dirname, 'clean.mjs' ), '--packages' ],
 			{ silent: true }
 		);
 
@@ -174,7 +173,7 @@ async function dev() {
 		// This must happen before TypeScript compilation because some packages
 		// (like vips) have source files that import from generated worker-code.ts
 		await exec( 'node', [
-			`"${ path.join( __dirname, 'packages/generate-worker-placeholders.mjs' ) }"`,
+			path.join( __dirname, 'packages/generate-worker-placeholders.mjs' ),
 		] );
 
 		if ( ! skipTypes ) {
@@ -191,16 +190,16 @@ async function dev() {
 
 			console.log( '\n✅ Checking type declaration files...' );
 			await exec( 'node', [
-				`"${ path.join(
+				path.join(
 					__dirname,
 					'packages/check-build-type-declaration-files.cjs'
-				) }"`,
+				),
 			] );
 		}
 
 		console.log( '\n📦 Building vendor files...' );
 		await exec( 'node', [
-			`"${ path.join( __dirname, 'packages/build-vendors.mjs' ) }"`,
+			path.join( __dirname, 'packages/build-vendors.mjs' ),
 		] );
 
 		const setupTime = Date.now() - startTime;
@@ -231,7 +230,6 @@ async function dev() {
 		const buildWatch = spawn( 'wp-build', [ '--watch' ], {
 			cwd: ROOT_DIR,
 			stdio: [ 'inherit', 'pipe', 'inherit' ],
-			shell: true,
 			env: { ...process.env, NODE_ENV: 'development' },
 		} );
 

